### PR TITLE
Support all resolutions

### PR DESCRIPTION
**Pull request by chrisoro**
- Refactored scaling to support all resolutions. Resolution scaling and black bars are calculated according to D4 specifications. Everything greater than 16 / 10 aspect ratio is supported. Lower than 16/10 will introduce black bars on top/bottom (easy to calc) but some strange ui shift which I'm not able to calculate. Since these res are very uncommon anyways, I don't think this is a problem
- Recalculate scaling every time a the d4 window is found/moved/resized/...
- Only read config from the user folder if the params.ini file there is not empty

### CODE REVIEW

File: src/template_finder.py [hunk 1]
```diff
@@ -106,7 +106,9 @@ def wait_for_update(img: np.ndarray, roi: list[int] = None, timeout: float = 3,
 def stored_templates() -> dict[Template]:
     paths = []
     templates = {}
-    for path in [f"assets\\templates_{Cam().res_p}"]:
+    # TODO check if it works for higher res
```
Comment (chrisoro) [inline src/template_finder.py:109]: must still check

**Comment by chrisoro** [comment]
@aeon0 for some reason some unit tests are failing. I double checked the implementation and it should be good. Can you please take a look? why some tests fail?

**Comment by chrisoro** [comment]
We could also keep the highest resolution template and then just scale it to the others. This would resolve the need for templates for all resolutions, right?

File: src/template_finder.py (hunk 1 above)
Comment (chrisoro) [reply src/template_finder.py:109]: it does work if it is close enough. But never worse than old implementation

**Comment by chrisoro** [comment]
All done

**Review by aeon0 — APPROVED** [review]
awesome!

**Pull request merged**